portal/app/core/templates/partials/main_home.html

```diff
@@ -20,6 +99,20 @@ <h3>Powernet Home</h3>
         window.localStorage.setItem('token', token);
         window.localStorage.setItem('homes', JSON.stringify(homeList));
         window.localStorage.setItem('devices', JSON.stringify(deviceList));
+
+        // skycons
+        var icons = new Skycons(),
+          list  = [
+            "clear-day", "clear-night", "partly-cloudy-day",
+            "partly-cloudy-night", "cloudy", "rain", "sleet", "snow", "wind",
+            "fog"
+          ],
+          i;
+
+        for(i = list.length; i--; )
+        icons.set(list[i], list[i]);
+
+        icons.play();
```

Comment:
Did you copy and paste this code from somewhere?
The formatting and patterns here not good. Please don't ever do this. But braces around everything that should have it and indentation looks wrong.

Reply:
I forked from this project - https://github.com/darkskyapp/skycons , I went on and cleaned these files